### ISSUE TITLE
cfn-lint does not report invalid value for StackSet's PermissionModel property

### ISSUE THREAD

**Issue by jamesoff** [aws-cloudformation/cfn-python-lint#1740]
*cfn-lint version: (`cfn-lint --version`)*
cfn-lint 0.37.1

*Description of issue.*
cfn-lint does not report an invalid value for AWS::CloudFormation::StackSet.PermissionModel

Please provide as much information as possible:
* Template linting issues:
  * Please provide a CloudFormation sample that generated the issue.
  * If present, please add links to the (official) documentation for clarification.
  * Validate if the issue still exists with the latest version of `cfn-lint` and/or the latest Spec files
* Feature request:
  * Please provide argumentation about the missing feature. Context is key!

Documentation for resource: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudformation-stackset.html#cfn-cloudformation-stackset-permissionmodel

Sample template which lints without errors but does not deploy:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: "Testing StackSets"
Parameters:
  Region2:
    Type: String
Resources:
  StackRegion2:
    Type: AWS::CloudFormation::StackSet
    Properties:
      StackSetName: !Ref AWS::StackName
      AdministrationRoleARN: !Sub 'arn:aws:iam::${AWS::AccountId}:role/AWSCloudFormationStackSetAdministrationRole'
      Description: 'Deploy stack in secondary region'
      ExecutionRoleName: AWSCloudFormationStackSetExecutionRole
      StackInstancesGroup:
        - DeploymentTargets:
            Accounts:
              - !Ref 'AWS::AccountId'
          Regions:
            - !Ref Region2
      TemplateURL: https://somebucket.s3.us-east-1.amazonaws.com/sometempate.yaml
      PermissionModel: SELF-MANAGED
```

The correct value for `PermissionModel` is `SELF_MANAGED` with an underscore.

The stack launches but fails to create, reporting `Model validation failed (#: #: only 1 subschema matches out of 2) #/PermissionModel: #: only 1 subschema matches out of 2 (#/PermissionModel)` for the StackSet resource.

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)
Done. No change.
